confidant/authnz/__init__.py

```diff
@@ -90,6 +100,19 @@ def user_type_has_privilege(user_type, privilege):
     return False
 
 
+def user_is_authorized(groupname):
+    # Fail open if:
+    # - the principal is not a human user
+    # - authz checks are disabled globally
+    # - OR there is no group specified
+    return (
+        not user_is_user_type('user')
+        or not app.config.get('USE_GROUPS')
+        or not groupname
+        or user_is_member(groupname)
+    )
```

Comment:
This should be implemented as an ACL check against actions of a resource: https://lyft.github.io/confidant/acls.html